leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -258,15 +323,14 @@ boolean getAndRecordLeaderPingResult(@Nullable Future<Boolean> pingFuture) throw
     }
 
     private Optional<PingableLeader> getSuspectedLeaderOverNetwork(String uuid) {
-        CompletionService<Entry<String, PingableLeader>> pingService = new ExecutorCompletionService<>(
-                executor);
+        MultiplexingCompletionService<PingableLeader, Entry<String, PingableLeader>> pingService
```

Comment:
also curious about why we need to create this every time?

Reply:
(As mentioned above, responses to separate calls to `getSuspectedLeaderOverNetwork` aren't generally interchangeable, so we need a fresh one.)